File: selenium/test/java/com/thoughtworks/selenium/corebased/TestSelect.java
package com.thoughtworks.selenium.corebased;

import com.thoughtworks.selenium.*;
import org.testng.annotations.*;
import static org.testng.Assert.*;
import java.util.regex.Pattern;

public class TestSelect extends SeleneseTestNgHelper {
	@Test public void testSelect() throws Exception {
		selenium.open("../tests/html/test_select.html");
		assertTrue(selenium.isSomethingSelected("theSelect"));
		assertEquals(selenium.getSelectedLabel("theSelect"), "Second Option");;
		selenium.select("theSelect", "index=4");
		verifyEquals(selenium.getSelectedLabel("theSelect"), "Fifth Option");
		verifyEquals(selenium.getSelectedIndex("theSelect"), "4");;
		verifyEquals(selenium.getSelectedLabel("theSelect"), "Fifth Option");
		verifyEquals(join(selenium.getSelectedLabels("theSelect"), ','), "Fifth Option");
		selenium.select("theSelect", "Third Option");
		verifyEquals(selenium.getSelectedLabel("theSelect"), "Third Option");;
		verifyEquals(selenium.getSelectedLabel("theSelect"), "Third Option");;
		verifyEquals(selenium.getSelectedLabel("theSelect"), "Third Option");
		selenium.select("theSelect", "label=Fourth Option");
		verifyEquals(selenium.getSelectedLabel("theSelect"), "Fourth Option");
		verifyEquals(selenium.getSelectedLabel("theSelect"), "Fourth Option");;
		selenium.select("theSelect", "value=option6");
		verifyEquals(selenium.getSelectedLabel("theSelect"), "Sixth Option");
		verifyEquals(selenium.getSelectedValue("theSelect"), "option6");
		verifyEquals(selenium.getSelectedValue("theSelect"), "option6");;
		selenium.select("theSelect", "value=");
		verifyEquals(selenium.getSelectedLabel("theSelect"), "Empty Value Option");
		selenium.select("theSelect", "id=o4");
		verifyEquals(selenium.getSelectedLabel("theSelect"), "Fourth Option");
		verifyEquals(selenium.getSelectedId("theSelect"), "o4");
		selenium.select("theSelect", "");
		verifyEquals(selenium.getSelectedLabel("theSelect"), "");
		verifyEquals(join(selenium.getSelectedLabels("theSelect"), ','), "");
		try { selenium.select("theSelect", "Not an option"); fail("expected failure"); } catch (Throwable e) {}
		try { selenium.addSelection("theSelect", "Fourth Option"); fail("expected failure"); } catch (Throwable e) {}
		try { selenium.removeSelection("theSelect", "Fourth Option"); fail("expected failure"); } catch (Throwable e) {}
		verifyEquals(join(selenium.getSelectOptions("theSelect"), ','), "First Option,Second Option,Third Option,Fourth Option,Fifth Option,Sixth Option,Empty Value Option,");
	}
}

File: dotnet/test/webdriverbackedselenium/TestSelect.cs
using System;
using NUnit.Framework;

namespace Selenium.Tests
{
    [TestFixture]
    public class TestSelect : SeleniumTestCaseBase
    {
        [Test]
        public void ShouldBeAbleToSelect()
        {
            selenium.Open("../tests/html/test_select.html");
            Assert.IsTrue(selenium.IsSomethingSelected("theSelect"));
            Assert.AreEqual(selenium.GetSelectedLabel("theSelect"), "Second Option"); ;
            selenium.Select("theSelect", "index=4");
            Assert.AreEqual(selenium.GetSelectedLabel("theSelect"), "Fifth Option");
            Assert.AreEqual(selenium.GetSelectedIndex("theSelect"), "4");
            Assert.AreEqual(selenium.GetSelectedLabel("theSelect"), "Fifth Option");
            Assert.AreEqual(string.Join(",", selenium.GetSelectedLabels("theSelect")), "Fifth Option");
            selenium.Select("theSelect", "Third Option");
            Assert.AreEqual(selenium.GetSelectedLabel("theSelect"), "Third Option"); ;
            Assert.AreEqual(selenium.GetSelectedLabel("theSelect"), "Third Option"); ;
            Assert.AreEqual(selenium.GetSelectedLabel("theSelect"), "Third Option");
            selenium.Select("theSelect", "label=Fourth Option");
            Assert.AreEqual(selenium.GetSelectedLabel("theSelect"), "Fourth Option");
            Assert.AreEqual(selenium.GetSelectedLabel("theSelect"), "Fourth Option"); ;
            selenium.Select("theSelect", "value=option6");
            Assert.AreEqual(selenium.GetSelectedLabel("theSelect"), "Sixth Option");
            Assert.AreEqual(selenium.GetSelectedValue("theSelect"), "option6");
            Assert.AreEqual(selenium.GetSelectedValue("theSelect"), "option6"); ;
            selenium.Select("theSelect", "value=");
            Assert.AreEqual(selenium.GetSelectedLabel("theSelect"), "Empty Value Option");
            selenium.Select("theSelect", "id=o4");
            Assert.AreEqual(selenium.GetSelectedLabel("theSelect"), "Fourth Option");
            Assert.AreEqual(selenium.GetSelectedId("theSelect"), "o4");
            selenium.Select("theSelect", "");
            Assert.AreEqual(selenium.GetSelectedLabel("theSelect"), "");
            Assert.AreEqual(string.Join(",", selenium.GetSelectedLabels("theSelect")), "");
            try
            {
                selenium.Select("theSelect", "Not an option");
                Assert.Fail("expected failure");
            }
            catch (Exception)
            {
            }
            try
            {
                selenium.AddSelection("theSelect", "Fourth Option");
                Assert.Fail("expected failure");
            }
            catch (Exception)
            {
            }
            try
            {
                selenium.RemoveSelection("theSelect", "Fourth Option");
                Assert.Fail("expected failure");
            }
            catch (Exception)
            {
            }
            Assert.AreEqual(string.Join(",", selenium.GetSelectOptions("theSelect")),
                "First Option,Second Option,Third Option,Fourth Option,Fifth Option,Sixth Option,Empty Value Option,");
        }
    }
}
